utils/threadsafe/int32.go

```diff
@@ -0,0 +1,31 @@
+package threadsafe
```

Comment:
using `atomic` package might give us better performance as it doesn't use locks.
https://pkg.go.dev/sync/atomic#Value

Reply:
Performance depends on the number of running validators, therefore adding `atomic` doesn't benefit much.